samplers/aws/xray/internal/rule.go

```diff
@@ -0,0 +1,214 @@
+		sd = sdktrace.TraceIDRatioBased(r.ruleProperties.FixedRate).ShouldSample(parameters)
+
+		if sd.Decision == sdktrace.RecordAndSample {
+			atomic.AddInt64(&r.samplingStatistics.sampledRequests, int64(1))
+		}
+
+		return sd
+	}
+
+	// Take from reservoir quota, if quota is available for that second
+	if r.reservoir.take(now, false, 1.0) {
+		atomic.AddInt64(&r.samplingStatistics.sampledRequests, int64(1))
+		sd.Decision = sdktrace.RecordAndSample
+
+		return sd
+	}
+
+	// using traceIDRatioBased sampler to sample using fixed rate
+	sd = sdktrace.TraceIDRatioBased(r.ruleProperties.FixedRate).ShouldSample(parameters)
+
+	if sd.Decision == sdktrace.RecordAndSample {
+		atomic.AddInt64(&r.samplingStatistics.sampledRequests, int64(1))
+	}
+
+	return sd
+}
+
+// appliesTo performs a matching against rule properties to see
+// if a given rule does match with any of the rule set on AWS X-Ray console
+func (r *Rule) appliesTo(parameters sdktrace.SamplingParameters, serviceName string, cloudPlatform string) (bool, error) {
+	var httpTarget string
+	var httpURL string
+	var httpHost string
+	var httpMethod string
+	var HTTPURLPathMatcher bool
+
+	if parameters.Attributes != nil {
+		for _, attrs := range parameters.Attributes {
+			if attrs.Key == "http.target" {
+				httpTarget = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.url" {
+				httpURL = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.host" {
+				httpHost = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.method" {
+				httpMethod = attrs.Value.AsString()
+			}
+		}
+	}
+
+	// attributes and other HTTP span attributes matching
+	attributeMatcher, err := r.attributeMatching(parameters)
+	if err != nil {
+		return attributeMatcher, err
+	}
+	serviceNameMatcher, err := wildcardMatch(r.ruleProperties.ServiceName, serviceName)
+	if err != nil {
+		return serviceNameMatcher, err
+	}
+	serviceTypeMatcher, err := wildcardMatch(r.ruleProperties.ServiceType, cloudPlatform)
+	if err != nil {
+		return serviceTypeMatcher, err
+	}
+	HTTPMethodMatcher, err := wildcardMatch(r.ruleProperties.HTTPMethod, httpMethod)
+	if err != nil {
+		return HTTPMethodMatcher, err
+	}
+	HTTPHostMatcher, err := wildcardMatch(r.ruleProperties.Host, httpHost)
+	if err != nil {
+		return HTTPHostMatcher, err
+	}
+
+	if httpURL != "" {
+		HTTPURLPathMatcher, err = wildcardMatch(r.ruleProperties.URLPath, httpURL)
+		if err != nil {
+			return HTTPURLPathMatcher, err
+		}
+	} else {
+		HTTPURLPathMatcher, err = wildcardMatch(r.ruleProperties.URLPath, httpTarget)
+		if err != nil {
+			return HTTPURLPathMatcher, err
+		}
+	}
+
+	return attributeMatcher &&
+		serviceNameMatcher &&
+		serviceTypeMatcher &&
+		HTTPMethodMatcher &&
+		HTTPHostMatcher &&
+		HTTPURLPathMatcher, nil
+}
+
+// attributeMatching performs a match on attributes set by users on AWS X-Ray console
+func (r *Rule) attributeMatching(parameters sdktrace.SamplingParameters) (match bool, err error) {
+	match = false
```

Comment:
```suggestion
```

This is the zero value.  Though I'm not sure a named return value is needed here, so maybe remove that and instead:

```suggestion
	match := false
```